src/Symfony/Component/Cache/DataCollector/CacheDataCollector.php

```diff
@@ -0,0 +1,159 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\DataCollector;
+
+use Symfony\Component\Cache\Adapter\RecordingAdapter;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+/**
+ * @author Aaron Scherer <aequasi@gmail.com>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CacheDataCollector extends DataCollector
+{
+    /**
+     * @var RecordingAdapter[]
+     */
+    private $instances = array();
+
+    /**
+     * @param string           $name
+     * @param RecordingAdapter $instance
+     */
+    public function addInstance($name, RecordingAdapter $instance)
+    {
+        $this->instances[$name] = $instance;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+        $empty = array('calls' => array(), 'config' => array(), 'options' => array(), 'statistics' => array());
+        $this->data = array('instances' => $empty, 'total' => $empty);
+        foreach ($this->instances as $name => $instance) {
+            $this->data['instances']['calls'][$name] = $instance->getCalls();
+        }
+
+        $this->data['instances']['statistics'] = $this->calculateStatistics();
+        $this->data['total']['statistics'] = $this->calculateTotalStatistics();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'cache';
+    }
+
+    /**
+     * Method returns amount of logged Cache reads: "get" calls.
+     *
+     * @return array
+     */
+    public function getStatistics()
+    {
+        return $this->data['instances']['statistics'];
+    }
+
+    /**
+     * Method returns the statistic totals.
+     *
+     * @return array
+     */
+    public function getTotals()
+    {
+        return $this->data['total']['statistics'];
+    }
+
+    /**
+     * Method returns all logged Cache call objects.
+     *
+     * @return mixed
+     */
+    public function getCalls()
+    {
+        return $this->data['instances']['calls'];
+    }
+
+    /**
+     * @return array
+     */
+    private function calculateStatistics()
+    {
+        $statistics = array();
+        foreach ($this->data['instances']['calls'] as $name => $calls) {
+            $statistics[$name] = array(
+                'calls' => 0,
+                'time' => 0,
+                'reads' => 0,
+                'hits' => 0,
+                'misses' => 0,
+                'writes' => 0,
+                'deletes' => 0,
+            );
+            foreach ($calls as $call) {
+                $statistics[$name]['calls'] += 1;
+                $statistics[$name]['time'] += $call->time;
+                if ($call->name === 'getItem') {
```

Comment:
Thank you. If one do getItems('a', 'b', 'c'). I will count that as 1 call and 3 reads.